File: 0941-valid-mountain-array/0941-valid-mountain-array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int i = 0;
        while (i + 1 < arr.size() && arr[i] < arr[i + 1]) {
            i++;
        }
        int j = arr.size() - 1;
        while (j - 1 >= 0 && arr[j] < arr[j - 1]) {
            j--;
        }
        
        if(i == 0) return false;
        
        if(j == arr.size() - 1) return false;
        
        return i == j;
    }
};

File: 0941-valid-mountain-array/0941-valid-mountain-array.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        int i = 0; 
        int j = arr.length - 1;
        while (i + 1 < arr.length && arr[i + 1] > arr[i]) {
            i++;
        }
        while (j - 1 >=0 && arr[j - 1] > arr[j]) {
            j--;
        }
        return i > 0 && i == j && j < arr.length - 1;
    }
}